### PR TITLE
coalaCITest.py: Add --bear-dirs test

### PR DESCRIPTION
**Pull request by nvzard**
Add --bear-dirs setting
unit tests

Closes https://github.com/coala/coala/issues/4888

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: tests/coalaCITest.py [hunk 1]
```diff
@@ -172,3 +172,19 @@ def test_limit_files_negative(self):
...
+    def test_bear_dirs(self):
+        with bear_test_module(), \
+                prepare_file(['random_text'], None) as (lines, filename):
+            retval, stdout, stderr = execute_coala(
+                    coala.main, 'coala', '--non-interactive',
+                    '-c', os.devnull,
+                    '--bear-dir', 'test_bears',
+                    '-b', 'LineCountTestBear',
+                    '-f', filename,)
+            self.assertIn('This file has 1 lines.',
```
Comment (kriti21) [inline tests/coalaCITest.py]: The message in assert should be "This file has 1 **line** "

Comment (nvzard) [reply tests/coalaCITest.py]: Why?

Comment (kriti21) [reply tests/coalaCITest.py]: Oh yes. I am sorry i forgot about that.

File: tests/coalaCITest.py [hunk 2]
```diff
@@ -172,3 +172,19 @@ def test_limit_files_negative(self):
...
                              'coala must return zero when successful')
+
+    def test_bear_dirs(self):
+        with bear_test_module(), \
+                prepare_file(['random_text'], None) as (lines, filename):
+            retval, stdout, stderr = execute_coala(
+                    coala.main, 'coala', '--non-interactive',
+                    '-c', os.devnull,
+                    '--bear-dir', 'test_bears',
+                    '-b', 'LineCountTestBear',
```
Comment (sangamcse) [inline tests/coalaCITest.py]: I think there should be `LineCountBearTest`.

Comment (nvzard) [reply tests/coalaCITest.py]: @sangamcse  what do you mean o.O ? Name of the bear is LineCountTestBear ;)

Comment (sangamcse) [reply tests/coalaCITest.py]: Are you sure? Is this https://github.com/coala/coala-bears/blob/master/tests/general/LineCountBearTest.py you are talking about?

Comment (sangamcse) [reply tests/coalaCITest.py]: Cool. Got it. 😉

**Comment by nvzard** [comment]
@userzimmermann can you please take a look at this : )

File: tests/coalaCITest.py [hunk 3]
```diff
@@ -172,3 +172,19 @@ def test_limit_files_negative(self):
             self.assertFalse(stderr)
             self.assertEqual(retval, 0,
                              'coala must return zero when successful')
+
+    def test_bear_dirs(self):
+        with bear_test_module(), \
```
Comment (jayvdb) [inline tests/coalaCITest.py]: this already adds the test_bears directory, so the test logic below doesnt prove anything

Comment (nvzard) [reply tests/coalaCITest.py]: Made the changes :+1:  that's for pointing out.

Comment (jayvdb) [reply tests/coalaCITest.py]: but you are still using this context manager ... why ?

Comment (nvzard) [reply tests/coalaCITest.py]: Shizz, sorry for that. I removed the context manager now the test_bears dir will be added by --bear-dirs flag and we won't be needing the `HiddenTestBear`

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5144/commits/068369ff054fe4506fc9df5f86d72ccb47603e9e

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**